**Group 7**

ilin27

GitHub Link: https://github.com/ilin27/project_planning_stage_individual.git 
-

In [18]:
# Loading datasets into R
library(tidyverse)

url_players = "https://raw.githubusercontent.com/ilin27/project_planning_stage_individual/refs/heads/main/players.csv"
download.file(url_players, "players.csv")
players <- read_csv("players.csv")

url_sessions <- "https://raw.githubusercontent.com/ilin27/project_planning_stage_individual/refs/heads/main/sessions.csv"
download.file(url_sessions, destfile = "sessions.csv")
sessions <- read_csv("sessions.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
Rows: 1535 Columns: 5
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (3): hashedEmail, start_time, end_time
dbl (2): original_start_time, original_end_time

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [11]:
summary(players)

  experience        subscribe       hashedEmail         played_hours    
 Length:196         Mode :logical   Length:196         Min.   :  0.000  
 Class :character   FALSE:52        Class :character   1st Qu.:  0.000  
 Mode  :character   TRUE :144       Mode  :character   Median :  0.100  
                                                       Mean   :  5.846  
                                                       3rd Qu.:  0.600  
                                                       Max.   :223.100  
                                                                        
     name              gender               Age       
 Length:196         Length:196         Min.   : 9.00  
 Class :character   Class :character   1st Qu.:17.00  
 Mode  :character   Mode  :character   Median :19.00  
                                       Mean   :21.14  
                                       3rd Qu.:22.75  
                                       Max.   :58.00  
                               

In [12]:
summary(sessions)

 hashedEmail         start_time          end_time         original_start_time
 Length:1535        Length:1535        Length:1535        Min.   :1.712e+12  
 Class :character   Class :character   Class :character   1st Qu.:1.716e+12  
 Mode  :character   Mode  :character   Mode  :character   Median :1.719e+12  
                                                          Mean   :1.719e+12  
                                                          3rd Qu.:1.722e+12  
                                                          Max.   :1.727e+12  
                                                                             
 original_end_time  
 Min.   :1.712e+12  
 1st Qu.:1.716e+12  
 Median :1.719e+12  
 Mean   :1.719e+12  
 3rd Qu.:1.722e+12  
 Max.   :1.727e+12  
 NA's   :2          

**(1) Data Description:**
-

players.csv
=
- **Number of observations**: 196
- **Summary statistics (2 d.p.)**: Please refer to the tables displayed by summary(players) above. Some interesting ones to note:
    - Many more players subscribed (144) than not (52)
    - The mean hours played is about 5.8 hours
    - The average age of a player is 21 years old
- **Number of variables**: 7
    - experience <chr> The experience level of the player (Amateur, Regular, Pro, Veteran)
    - subscribe <lgl> Whether players have subscribed to the newsletter (TRUE, FALSE)
    - hashedEmail <chr>	The player's email, hashed for privacy
    - played_hours <dbl> The number of hours the player has spent on the server
    - name <chr> The player's name
    - gender <chr> The gender of the player (Male, Female, Non-binary, Agender, Two-Spirited, Prefer not to say)
    - Age <dbl> (The "A" is capitalized) The players age
- **Any issues in the data**: Since many more players have subscribed than not (higher percentage of TRUE), this may be an imbalance, affecting classification (something to keep in mind).
- **How the data was collected**: Dr. Frank Wood collected data on player actions as they navigate in his MineCraft server.

sessions.csv
=
- **Number of observations**: 1535
- **Summary statistics (2 d.p.)**: Please refer to the tables displayed by summary(sessions) above. Some interesting ones to note:
    - Since start_time and end_time are chr and not dbl, we cannot see useful summary information (e.g. average start time)
- **Number of variables**: 5
    - hashedEmail <chr> The player's email, hashed for privacy
    - start_time <chr> The start date and start time (format: DD/MM/YYYY HH:MM)
    - end_time <chr> The end date and end time (format: DD/MM/YYYY HH:MM)
    - original_start_time <dbl> The original start time (unix standard time) - meaningless in the context of the proposed question
    - original_end_time <dbl> The original end time (unix standard time) - meaningless in the context of the proposed question
- **Any issues in the data**: The start_time and end_time columns are not tidy, since there are two values (data and time) in each cell.
- **How the data was collected**: Dr. Frank Wood collected data on player login and logout times on his MineCraft server.

**(2) Questions:**
-

I will be exploring Question 1: **"What player characteristics and behaviours are most predictive of subscribing to a game-related newsletter, and how do these features differ between various player types?"**

Specific question: **"Can the number of hours played (played_hours) predict if a player will subscribe (subscribe) to a game-related newletter in the players.csv dataset?"**

The rationale for choosing the above predictor is that a greater number of hours played may indicate that the player has an interest in the game, and therefore would subscribe to the newsletter.

I will be focusing on the players.csv dataset, and will not be using the sessions.csv dataset since it does not contain any data relevant to the above stated question.

**(3) Exploratory Data Analysis and Visualization**
-

In [20]:
# Loading datasets into R
library(tidyverse)
url_players = "https://raw.githubusercontent.com/ilin27/project_planning_stage_individual/refs/heads/main/players.csv"
download.file(url_players, "players.csv")
players <- read_csv("players.csv")

Rows: 196 Columns: 7
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr (4): experience, hashedEmail, name, gender
dbl (2): played_hours, Age
lgl (1): subscribe

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.


In [21]:
## Wrangling

# The players.csv dataset is tidy. 
# Each row is a singular observation and each column is singular variable.
# Each cell contains a single value.
# Therefore, there is no need for minimum necessary wrangling to be done at this point of the project.

In [32]:
# Compute the mean value for each quantitative variable in the players.csv data set. 

summarize(players,
          mean_played_hours = mean(played_hours, na.rm = TRUE),
          mean_Age = mean(Age, na.rm = TRUE))

mean_played_hours mean_Age
1 5.845918          21.13918

In [23]:
# https://www.codecademy.com/resources/docs/markdown/tables

Table: Mean Quantitative Variables in players.csv
-
| mean_played_hours    | mean_Age |
| -------------------- | -------- |
| 5.846                | 21       |

In [24]:
# A few exploratory visualizations of the data.
# Make sure to include labels, titles, units of measurement, etc.
# Explain any insights you gain from these plots that are relevant to address your question.